Library/Homebrew/download_strategy.rb

```diff
@@ -269,6 +269,23 @@ def initialize(name, resource)
 
   def fetch
     ohai "Downloading #{@url}"
+
+    begin
+      does_redirect
+
+      if !does_redirect.include?(@url) && ENV["HOMEBREW_NO_INSECURE_REDIRECT"].nil?
+        ohai "Downloaded from #{does_redirect}".gsub("Location: ", "")
+      elsif @url.include?("https") && does_redirect.include?("http")
+        puts "Insecure redirect detected! Escaping to safety."
+        exit! 1
```

Comment:
Probably wants to be an exception (see others in this file) rather than an `exit`